HTTPieCodeGenerator.coffee

```diff
@@ -111,8 +111,51 @@ HTTPieCodeGenerator = ->
                 s = ""
                 for key, value of object
                     sign = "#{if typeof(value) == 'string' then "=" else ":="}"
-                    s += "    #{addslashes key}#{sign}#{@json_body_object(value)} \\\n"
+                    if typeof value == 'object'
+                      s += addslashes(key) + sign
+                      if value.length != null
+                        s += @nested_array_object(value, 1)
+                    else
+                      s += '\'{\n' + @nested_hash_object(value, 1) + '}\''
         return s
+        
+    @nested_array_object = (array, level) ->
```

Comment:
@matrinox why doing `nested_array_object` and `nested_hash_object` manually?
this would work nicely `JSON.stringify(object, null, '  ')` I guess…
I'm working on a fix for this already, I just wanted to hear your opinion. Thanks!